lib/gui/models/settings.js

```diff
@@ -34,12 +36,25 @@ const Store = require('./store');
  * settings.set('unmountOnSuccess', true);
  */
 exports.set = (key, value) => {
+  if (_.isNil(key)) {
+    throw errors.createError({
+      title: 'Missing setting key'
+    });
+  }
+
+  if (!_.isString(key)) {
+    throw errors.createError({
+      title: `Invalid setting key: ${key}`
+    });
+  }
+
+  const newSettings = _.assign(exports.getAll(), _.fromPairs([
+    [ key, value ]
```

Comment:
I was looking at this line, and suspected there must be a simpler way :+1: 

Ah no wait, it was https://github.com/resin-io/etcher/pull/1515/files#diff-74bcbd72e100aecd2f68e833e7a169f4R71 that I was suspecting could be simplified.